src/precomputed_quantities.jl

```diff
@@ -59,20 +59,104 @@ function precomputed_quantities(atmos, center_space, face_space)
 end
 
 """
-    divide_by_ρa(ρaχ, ρa, ρχ, ρ, a_min)
+    divide_by_ρa_anelastic(ρaχʲ, ρaʲ, ρχ, ρ, a_half)
 
-Computes `ρaχ / ρa`, regularizing the result to avoid issues when `a` is small.
-This is done by performing a linear interpolation from `ρaχ / ρa` to `ρχ / ρ`,
-interpolating closer to `ρχ / ρ` when `a` is small. When `a == 0`, the result is
-exactly `ρχ / ρ` (a tiny value of `ε` is added to the denominator of `ρaχ / ρa`
-in order to avoid returning `NaN` when `a == 0`, since `0 * Inf` is `NaN`).
+Calls `divide_by_ρa` with `aʲ = ρaʲ / ρ` (i.e., using the anelastic
+approximation to compute `aʲ`). This is required for computing specific SGS
+quantities before the value of `aʲ` has been computed using the SGS density,
+which is obtained from the SGS thermodynamic state.
 """
-function divide_by_ρa(ρaχ, ρa, ρχ, ρ, a_min)
-    ε = eps(zero(ρa)) # This is the smallest positive number we can add to ρa.
-    sgs_weight = 1 / (1 + a_min^2 * ρ^2 / ρa^2) # TODO: Make this exponential.
-    return sgs_weight * ρaχ / (ρa + ε) + (1 - sgs_weight) * ρχ / ρ
+divide_by_ρa_anelastic(ρaχʲ, ρaʲ, ρχ, ρ, a_half) =
+    divide_by_ρa(ρaχʲ, ρaʲ, ρχ, ρ, ρaʲ / ρ, a_half)
+
+"""
+    divide_by_ρa(ρaχʲ, ρaʲ, ρχ, ρ, aʲ, a_half)
+
+Computes `ρaχʲ / ρaʲ`, regularizing the result to avoid issues when `aʲ` is
+small. This is done by performing a linear interpolation from `ρaχʲ / ρaʲ` to
+`ρχ / ρ`, using the `sgs_weight_function` as the weight of `ρaχʲ / ρaʲ`.
+"""
+function divide_by_ρa(ρaχʲ, ρaʲ, ρχ, ρ, aʲ, a_half)
+    weight = sgs_weight_function(aʲ, a_half)
+    # If ρaʲ = 0, we know that aʲ = 0 (since ρʲ cannot be 0), which means that
+    # weight = 0. However, 0 * ρaχʲ / 0 = NaN, regardless of what ρaχʲ is, so
+    # the linear interpolation will always return NaN when ρaʲ = 0. To avoid
+    # this problem, we need to have a special case for ρaʲ = 0.
+    return ρaʲ == 0 ? ρχ / ρ : weight * ρaχʲ / ρaʲ + (1 - weight) * ρχ / ρ
 end
 
+"""
+    sgs_weight_function(aʲ, a_half)
+
+Computes the weight of the SGS variables in the linear interpolation used in
+`divide_by_ρa`. This is a continuously differentiable and monotonically
+increasing function of `aʲ` that is equal to 0 when `aʲ ≤ 0`, is equal to 1 when
+`aʲ ≥ 1`, is equal to `1 / 2` when `aʲ = a_half`, grows very rapidly near
+`aʲ = a_half`, and grows very slowly at all other values of `aʲ`.  If `a_half`
+is sufficiently small, this function is essentially equal to 1 for all `aʲ` more
+than a few times larger than `a_half` (up to floating-point precision).
+
+Here is a brief description of how this function is constructed. We need the
```

Comment:
Would the rest of this long dosctring be more suited for documentation page?

Reply:
This way we could include a plot of the function, which is probably better than trying to describe what is happening with words